aws/resource_aws_sagemaker_lifecycle_configuration.go

```diff
@@ -0,0 +1,168 @@
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"on_create": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 16384),
+			},
+
+			"on_start": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 16384),
+			},
+		},
+	}
+}
+
+func resourceAwsSagemakerLifeCycleConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	var name string
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+	} else {
+		name = resource.UniqueId()
+	}
+
+	createOpts := &sagemaker.CreateNotebookInstanceLifecycleConfigInput{
+		NotebookInstanceLifecycleConfigName: aws.String(name),
+	}
+
+	// on_create is technically a list of NotebookInstanceLifecycleHook elements, but the list has to be length 1
+	// (same for on_start)
+	if v, ok := d.GetOk("on_create"); ok {
+		hook := &sagemaker.NotebookInstanceLifecycleHook{Content: aws.String(v.(string))}
+		createOpts.SetOnCreate([]*sagemaker.NotebookInstanceLifecycleHook{hook})
+	}
+
+	if v, ok := d.GetOk("on_start"); ok {
+		hook := &sagemaker.NotebookInstanceLifecycleHook{Content: aws.String(v.(string))}
+		createOpts.SetOnStart([]*sagemaker.NotebookInstanceLifecycleHook{hook})
+	}
+
+	log.Printf("[DEBUG] SageMaker notebook instance lifecycle configuration create config: %#v", *createOpts)
+	_, err := conn.CreateNotebookInstanceLifecycleConfig(createOpts)
+	if err != nil {
+		return fmt.Errorf("error creating SageMaker notebook instance lifecycle configuration: %s", err)
+	}
+	d.SetId(name)
+
+	return resourceAwsSagemakerLifeCycleConfigurationRead(d, meta)
+}
+
+func resourceAwsSagemakerLifeCycleConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	request := &sagemaker.DescribeNotebookInstanceLifecycleConfigInput{
+		NotebookInstanceLifecycleConfigName: aws.String(d.Id()),
+	}
+
+	lifecycleConfig, err := conn.DescribeNotebookInstanceLifecycleConfig(request)
+	if err != nil {
+		if isAWSErr(err, "ValidationException", "") {
+			log.Printf("[INFO] unable to find the SageMaker notebook instance lifecycle configuration (%s); therefore it is removed from the state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error reading SageMaker notebook instance lifecycle configuration %s: %s", d.Id(), err)
+	}
+
+	if err := d.Set("name", lifecycleConfig.NotebookInstanceLifecycleConfigName); err != nil {
+		return fmt.Errorf("error setting name for SageMaker notebook instance lifecycle configuration (%s): %s", d.Id(), err)
+	}
+
+	if len(lifecycleConfig.OnCreate) > 0 {
+		if err := d.Set("on_create", lifecycleConfig.OnCreate[0].Content); err != nil {
+			return fmt.Errorf("error setting on_create for SageMaker notebook instance lifecycle configuration (%s): %s", d.Id(), err)
+		}
+	}
+
+	if len(lifecycleConfig.OnStart) > 0 {
```

Comment:
To prevent potential panics, this should perform a `nil` check

```suggestion
	if len(lifecycleConfig.OnStart) > 0 && lifecycleConfig.OnStart[0] != nil {
```